library/core/src/ptr/const_ptr.rs

```diff
@@ -645,6 +646,13 @@ impl<T: ?Sized> *const T {
     /// (Note that [`offset`] and [`add`] also have a similar limitation and hence cannot be used on
     /// such large allocations either.)
     ///
+    /// The requirement for pointers to be derived from the same allocated object is primarily
+    /// needed for `const`-compatibility: at compile-time, pointers into *different* allocated
+    /// objects do not have a known distance to each other. However, the requirement also exists at
```

Comment:
Reviewing this again... Would it be correct to strengthen this claim to something more like "allocations don't necessarily _have_ addresses during CTFE, just offsets from their base, so there is *no such thing* as a distance between two allocations"?

I know it amounts to the same, but I feel like "do not have a known distance" is slightly ambiguous, and this is basically my understanding of how it works. We should be clearer that the semantics of CTFE are allowed to be _incomprehensible_ where we can, to dissuade anyone who imagines some "like a flat address space" model applies.

Reply:
Yeah both of these are correct ways to explain the situation. I don't know which one works better for most people.
We have existing guidance that is printed by the compiler when e.g. you try to transmute a pointer to an integer, which reads as follows:
```
this code performed an operation that depends on the underlying bytes representing a pointer
the absolute address of a pointer is not known at compile-time, so such operations are not supported
```
So I went with the same idea of "address is not known yet" here. We could also go with "the address has not been assigned yet" or something like that, but then we should do it consistently in the libs docs and compiler output, I think.

Reply:
Yeah, doing it consistently is a good point so that should be a separate issue probably.